mapie/quantile_regression.py

```diff
@@ -0,0 +1,323 @@
+from __future__ import annotations
+import warnings
+from typing import List, Optional, Tuple, Union, cast
+
+import numpy as np
+from sklearn.base import RegressorMixin, clone
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import QuantileRegressor
+from sklearn.model_selection import BaseCrossValidator
+from sklearn.utils.validation import (
+    indexable,
+    check_is_fitted,
+    # _num_samples,
+    _check_y,
+)
+
+from ._typing import ArrayLike, NDArray
+from .utils import (
+    check_alpha,
+    check_alpha_and_n_samples,
+    check_n_features_in,
+    # check_n_jobs,
+    # check_nan_in_aposteriori_prediction, <-- check what happens
+    # when you also include this!
+    check_null_weight,
+    # check_verbose,
+    fit_estimator,
+)
+from ._compatibility import np_quantile
+from .regression import MapieRegressor
+
+
+class MapieQuantileRegressor(MapieRegressor):
+    valid_methods_ = ["quantile"]
+    valid_agg_functions_ = [None, "median", "mean"]
+    fit_attributes = [
+        "list_estimators",
+        "list_y_preds_calib",
+        "conformity_scores_",
+        "y_calib"
+    ]
+
+    link_estimator_quantile = {
+        "GradientBoostingRegressor": ["loss", "alpha"],
+        "QuantileRegressor": ["quantile", "quantile"]
+    }
+
+    def __init__(
+        self,
+        estimator: RegressorMixin = None,
+        method: str = "quantile",
+        cv: Optional[Union[int, str, BaseCrossValidator]] = None,
```

Comment:
At the moment, `MapieQuantileRegressor` accepts `cv` of type `str` ("split" or "prefit").

Reply:
What would be the prefit method? For the moment only the split method has been implemented.

Reply:
The "prefit" method is when you already trained your base models on a train set and you provide the fitted models together with the calibration sets to Mapie. I don't think it is adapted for quantile regression since the user would have to feed three fitted models.

So `cv` accepts only "split".

Reply:
I would keep `cv` as `str` only since the typing does not aim at allowing for typing errors performed in the tests.

Reply:
done